app/models/government.rb

```diff
@@ -20,4 +22,51 @@ def self.on_date(date)
   def current?
     self == Government.current
   end
+
+  def overlaps?(other)
+    !(is_before?(other) || is_after?(other))
+  end
+
+private
+
+  def is_before?(other)
+    starts_before?(other) && ends_before?(other)
+  end
+
+  def is_after?(other)
+    starts_after?(other) && ends_after?(other)
+  end
+
+  def starts_before?(other)
+    start_date < other.start_date
+  end
+
+  def ends_before?(other)
+    (end_date || distant_future) < other.start_date
```

Comment:
I feel like we could achieve the same thing without having to have the idea of `distant_future`. For example, this could be rewritten as `end_date.present? && end_date < other.start_date`. Same for the other checks.